src/parser/core/modules/Defensives.tsx

```diff
@@ -1,89 +1,163 @@
+import {Accordion, Button, Icon, Message, Table} from 'semantic-ui-react'
+import {CooldownHistoryEntry, Cooldowns} from './Cooldowns'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+import {Timeline} from './Timeline'
+
+const DEFENSIVE_ROLE_ACTIONS: Map<RoleKey, ActionKey[]> = new Map<RoleKey, ActionKey[]>([
+	['TANK', ['RAMPART', 'REPRISAL']],
+	['MELEE', ['FEINT', 'BLOODBATH', 'SECOND_WIND']],
+	['PHYSICAL_RANGED', ['SECOND_WIND']],
+	['MAGICAL_RANGED', ['ADDLE']],
+	['HEALER', []],
+])
+
+export class Defensives extends Analyser {
 	static override handle = 'defensives'
 	static override title = t('core.defensives.title')`Defensives`
+	static override displayOrder = DISPLAY_ORDER.DEFENSIVES
 
+	@dependency protected cooldowns!:Cooldowns
 	@dependency protected cooldownDowntime!: CooldownDowntime
 	@dependency protected data!: Data
-	@dependency private statistics!: Statistics
+	@dependency private timeline!: Timeline
 
 	/**
-	 * Implementing modules MUST provide a list of defensive actions to track
+	 * Implementing modules should provide a list of job-specific defensive actions to track
 	 */
-	protected abstract trackedDefensives: Action[]
-
+	protected trackedDefensives: Action[] = []
 	/**
-	 * Implementing modules may provide opts for the statistic display
+	 * Implementing modules may override the main header message text
 	 */
-	protected statisticOpts: AbstractStatisticOptions = {}
-
-	private uses: Map<Action['id'], number> = new Map()
+	protected headerContent: ReactNode = <Trans id="core.defensives.header.content">
+		Using your mitigation and healing cooldowns can help you survive mistakes, or relieve some stress on the healers and let them deal more damage.<br/>
+		While you shouldn't use them at the expense of your rotation or buff alignment, you should try to find helpful times to use them.
+	</Trans>
 
 	override initialise() {
-		const actionFilter = filter<Event>()
-			.source(this.parser.actor.id)
-			.type('action')
-
-		this.trackedDefensives.forEach(defensive => {
-			this.addEventHook(actionFilter.action(defensive.id),
-				() => this.uses.set(defensive.id, (this.uses.get(defensive.id) ?? 0) + 1)
-			)
+		const roleDefensives = DEFENSIVE_ROLE_ACTIONS.get(JOBS[this.parser.actor.job].role)?.map(key => this.data.actions[key]) ?? []
+		roleDefensives.forEach(roleAction => {
+			if (!this.trackedDefensives.find(action => roleAction.id === action.id)) {
+				this.trackedDefensives.push(roleAction)
+			}
 		})
+	}
 
-		this.addEventHook('complete', this.onComplete)
+	private getUses(defensive: Action): number {
+		return this.cooldowns.cooldownHistory(defensive).length
 	}
 
-	private formatUsages(defensive: Action): React.ReactNode {
-		const uses = this.uses.get(defensive.id) ?? 0
-		const maxUses = this.cooldownDowntime.calculateMaxUsages({cooldowns: [defensive]})
+	private getMaxUses(defensive: Action): number {
+		return this.cooldownDowntime.calculateMaxUsages({cooldowns: [defensive]})
```

Comment:
This isn't accounting for drift required to match the existing usages, and results in showing a Max Uses that's higher than what we're recommending in the details.  See Physis II on this log - 3/8 uses shown, but the breakdown only shows 3 possible extra uses in the downtimes.
http://localhost:3000/fflogs/A948DrvXpx1MzRaL/5/6

Reply:
That is intentional. I want to both show where the user can get additional uses given their existing ones, and also show them whether or not they're truly maximizing the potential usage of the skill. Think of something linke Pneuma that's got a long CD. If someone were to consistently use it at an interval of 235s for example, we wouldn't be showing any extra opportunities since there's no way to fit an additional 120s CD in between, but they'd only be getting about half of the possible uses in a fight. If there's a better way to communicate that then I'm all ears, but the numbers are technically correct to what they should be.

Reply:
After our additional discussion on Discord, revising this to show the maximum as the sum of actual usage plus the extra uses shown in the table. I toyed with having the accordion labels be [ActionName] - Used: [count] - Available: [count] or something like that and it didn't feel better than the simple [used]/[total] version so I left it that way with the calculation fix in place